src/material/datepicker/testing/calendar-harness.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {HarnessPredicate, ComponentHarness} from '@angular/cdk/testing';
+import {CalendarHarnessFilters, CalendarCellHarnessFilters} from './datepicker-harness-filters';
+import {MatCalendarCellHarness} from './calendar-cell-harness';
+
+/** Possible views of a `MatCalendarHarness`. */
+export const enum CalendarView {MONTH, YEAR, MULTI_YEAR}
+
+/** Harness for interacting with a standard Material calendar in tests. */
+export class MatCalendarHarness extends ComponentHarness {
```

Comment:
Should this class have a way to select an option in the current view?

Reply:
The idea was that you'd do `calendar.getCells({text: '10'})[0].select()`, but I supposed that we can have a shorthand for it.

Reply:
That makes sense. Let's leave it as is and let's add a convenience method in a follow-up PR

Reply:
Yeah I think we added shortcut convenience methods for the most common actions on other components too, like menu and select. Follow-up PR sounds good though since this one is already quite large

Reply:
Added.